script/misc-helper.ts

```diff
@@ -332,31 +332,52 @@ async function getCIHelper(): Promise<CIHelper> {
         const glue = new GitHubGlue();
         await glue.addPRComment(pullRequestURL, comment);
     } else if (command === "set-app-token") {
-        if (commander.args.length !== 1) {
+        if (commander.args.length < 1) {
             process.stderr.write(`${command}: unexpected argument(s)\n`);
             process.exit(1);
         }
 
-        const githubAppID = 12836;
-        const githubAppInstallationID = 195971;
-
-        const client = new octokit();
-        const key = await gitConfig("gitgitgadget.privateKey");
-        if (!key) {
-            throw new Error(`Need the App's private key`);
-        }
-        const now = Math.round(Date.now() / 1000);
-        const payload = { iat: now, exp: now + 60, iss: githubAppID };
-        const signOpts = { algorithm: "RS256" };
-        const app = jwt.sign(payload, key.replace(/\\n/g, `\n`), signOpts);
-        client.authenticate({
-            token: app,
-            type: "app",
-        });
-        const result = await client.apps.createInstallationToken({
-            installation_id: githubAppInstallationID,
-        });
-        await git(["config", "gitgitgadget.githubToken", result.data.token]);
+        const set = async (options: {
+             appID: number,
+             installationID?: number,
+             name: string,
+        }): Promise<void> => {
+            const client = new octokit();
+            const appName = options.name === "gitgitgadget" ?
+                "gitgitgadget" : "gitgitgadget-git";
+            const key = await gitConfig(`${appName}.privateKey`);
+            if (!key) {
+                throw new Error(`Need the ${appName} App's private key`);
+            }
+            const now = Math.round(Date.now() / 1000);
+            const payload = { iat: now, exp: now + 60, iss: options.appID };
+            const signOpts = { algorithm: "RS256" };
+            const app = jwt.sign(payload, key.replace(/\\n/g, `\n`), signOpts);
+            client.authenticate({
+                token: app,
+                type: "app",
+            });
+
+            if (options.installationID === undefined) {
+                options.installationID =
+                    (await client.apps.getRepoInstallation({
+                        owner: options.name,
+                        repo: "git",
+                })).data.id;
+            }
+            const result = await client.apps.createInstallationToken({
+                installation_id: options.installationID!,
+            });
+            const configKey = options.name === "gitgitgadget" ?
```

Comment:
Why have different values for `configKey`?  ie why not always set to `gitgitgadget.${options.name}.githubToken`?  This would make it simpler in github-glue.ts?

Reply:
Legacy reasons. The Azure Pipeline uses the same token also for pushing, outside of the Typescript code, in a Bash script step.

Reply:
(And I do want that Azure Pipeline to continue working seamlessly both before and after this PR is merged.)